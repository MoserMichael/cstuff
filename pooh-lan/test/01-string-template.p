


sub foo()
  return 42
end

a  = 1

# string template with statement; this translates into an anonymous function
println( ~msg '  aaa [ global.a = 2 return 'a' ]' )


# string template with function call:
println( ~msg '  bbb [ 5 * foo() ] ' )

# the first template changed value of a global as a side effect.
println( ~msg a )
