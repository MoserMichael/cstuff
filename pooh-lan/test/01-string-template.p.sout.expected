  aaa a
  bbb 210 
2
